code/Controllers/ModelAsController.php

```diff
@@ -72,10 +72,8 @@ protected function beforeHandleRequest(HTTPRequest $request)
 
     /**
      * @uses ModelAsController::getNestedController()
-     * @param HTTPRequest $request
-     * @return HTTPResponse
      */
-    public function handleRequest(HTTPRequest $request)
+    public function handleRequest(HTTPRequest $request): HTTPResponse
```

Comment:
This sometimes returns a result directly from an arbitrary `RequestHandler::handleRequest()` which doesn't guarantee an `HTTPResponse` - probably the returned value needs the same treatment that `Controller::handleRequest()` does on those.

Reply:
I've added strong typing to `ModelAsController::controller_for():ContentController` to ensure teh method is always available

The phpdoc on that method says it's only supposed to only return a ContentController, so I'm happy for hard fails here on upgrade if something doesn't meet the new signature